From foo@baz Mon Jan  5 10:21:08 PST 2009
Date: Mon, 05 Jan 2009 10:21:08 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: go7007: fixes due to video_usercopy api change

video_usercopy() just changed its arguments, so fix up the go7007 driver
to properly build.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/go7007/go7007-v4l2.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/staging/go7007/go7007-v4l2.c
+++ b/drivers/staging/go7007/go7007-v4l2.c
@@ -375,8 +375,7 @@ static int clip_to_modet_map(struct go70
 	return 0;
 }
 
-static int go7007_do_ioctl(struct inode *inode, struct file *file,
-		unsigned int cmd, void *arg)
+static long go7007_do_ioctl(struct file *file, unsigned int cmd, void *arg)
 {
 	struct go7007_file *gofh = file->private_data;
 	struct go7007 *go = gofh->go;
@@ -1335,7 +1334,7 @@ static int go7007_ioctl(struct inode *in
 	if (gofh->go->status != STATUS_ONLINE)
 		return -EIO;
 
-	return video_usercopy(inode, file, cmd, arg, go7007_do_ioctl);
+	return video_usercopy(file, cmd, arg, go7007_do_ioctl);
 }
 
 static ssize_t go7007_read(struct file *file, char __user *data,
